ocaml/xapi/xapi_database_backup.ml

```diff
@@ -0,0 +1,175 @@
+module D=Debug.Make(struct let name="xapi_database_backup" end)
+open D
+
+open Http
+open Xmlrpc_client
+open Db_cache_types
+open Stdext
+open Threadext
+
+type stat = {created : int64; modified : int64; deleted : int64;} [@@deriving rpc]
+type field = {fldname : string; stat : stat; value : Schema.Value.t;} [@@deriving rpc]
+type row = {objref  : string; stat : stat; value : field list;} [@@deriving rpc]
+type table = {tblname : string; stat : stat; value : row list;} [@@deriving rpc]
+
+type del_tables = {key: string; table: string; stat : stat;} [@@deriving rpc]
+type delta = {fresh_token: int64; tables : table list; deletes : del_tables list;} [@@deriving rpc]
+type edits = {ts : TableSet.t; tables : string list}
+
+type count = {tblname:string; count: int; del:int;}
+type pair = {tbl:string; row:string;}
+
+let empty_schema = Schema.Value.unmarshal String ""
```

Comment:
Is this an empty-string value rather than an empty schema? You can probably just write `Schema.Value.String ""`?